lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlobFileSystem.cs

```diff
@@ -0,0 +1,255 @@
+                {
+                    throw;
+                }
+
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Copies to a CloudBlockBlob from another block blob.
+        /// <see cref="IFileSystem.Copy"/>
+        /// </summary>
+        public void Copy(Uri sourceUri, Uri destinationUri)
+        {
+            new CloudBlockBlob(destinationUri, _client.Credentials).StartCopyFromBlobAsync(sourceUri).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// <see cref="IFileSystem.CopyToLocal"/>
+        /// </summary>
+        public void CopyToLocal(Uri remoteFileUri, string localName)
+        {
+            _client.GetBlobReferenceFromServer(remoteFileUri).DownloadToFile(localName, FileMode.CreateNew);
+        }
+
+        /// <summary>
+        // Copies to a CloudBlockBlob from a local file.
+        /// <see cref="IFileSystem.CopyFromLocal"/>
+        /// </summary>
+        public void CopyFromLocal(string localFileName, Uri remoteFileUri)
+        {
+            new CloudBlockBlob(remoteFileUri, _client.Credentials).UploadFromFile(localFileName, FileMode.Open);
+        }
+
+        /// <summary>
+        /// Azure blobs does not have the sense of the existence of a "directory."
+        /// We will not throw an error, but we will not do anything either.
+        /// </summary>
+        public void CreateDirectory(Uri directoryUri)
+        {
+        }
+
+        /// <summary>
+        /// Recursively deletes blobs under a specified "directory URI."
+        /// If only the container is specified, the entire container is deleted.
+        /// </summary>
+        public void DeleteDirectory(Uri directoryUri)
+        {
+            var uriSplit = directoryUri.AbsolutePath.Split(new[] {"/"}, StringSplitOptions.RemoveEmptyEntries);
+            if (!uriSplit.Any())
+            {
+                throw new ArgumentOutOfRangeException(@"URI must contain at least the container");
+            }
+
+            var containerName = uriSplit[0];
+            var container = _client.GetContainerReference(containerName);
+            if (uriSplit.Length == 1)
+            {
+                container.DeleteIfExists();
+                return;
+            }
+
+            var directory = container.GetDirectoryReference(uriSplit[1]);
+
+            for (var i = 2; i < uriSplit.Length; i++)
+            {
+                directory = directory.GetDirectoryReference(uriSplit[i]);
+            }
+
+            foreach (var blob in directory.ListBlobs(true).OfType<ICloudBlob>())
+            {
+                blob.DeleteIfExists();
+            }
+        }
+
+        /// <summary>
+        /// Gets the children of the blob "directory."
+        /// </summary>
+        public IEnumerable<Uri> GetChildren(Uri directoryUri)
+        {
+            BlobContinuationToken blobContinuationToken = null;
+            var path = directoryUri.AbsolutePath.Trim('/');
```

Comment:
https://msdn.microsoft.com/en-us/library/azure/jj682530.aspx
The call asks for the prefix, but doesn't ask for the absolute URI.